fondant/component.py

```diff
@@ -223,19 +225,105 @@ def transform(self, *args, **kwargs) -> dd.DataFrame:
 
     def _process_dataset(self, manifest: Manifest) -> t.Union[None, dd.DataFrame]:
         """
-        Creates a DataLoader using the provided manifest and loads the input dataframe using the
-        `load_dataframe` instance, and  applies data transformations to it using the `transform`
-        method implemented by the derived class. Returns a single dataframe.
+        Load the data based on the manifest using a DaskDataloader and call the transform method to
+        process it.
+
+        Returns:
+            A `dd.DataFrame` instance with updated data based on the applied data transformations.
+        """
+
+
+class DaskTransformComponent(TransformComponent):
+    @abstractmethod
+    def transform(self, *args, **kwargs) -> dd.DataFrame:
+        """
+        Abstract method for applying data transformations to the input dataframe.
+
+        Args:
+            args: A Dask dataframe will be passed in as a positional argument
+            kwargs: Arguments provided to the component are passed as keyword arguments
+        """
+
+    def _process_dataset(self, manifest: Manifest) -> dd.DataFrame:
```

Comment:
the return type seems to differ from the one defined in the abstract method `t.Union[None, dd.DataFrame]`

Reply:
I don't think that's an issue. The `None` is only for the `WriteComponent`, and the `dd.DataFrame` for the `Transform` and `LoadComponent`. I updated it everywhere to reflect this.

Reply:
They shouldn't return a mismatch when returning a stricter value, only when accepting a stricter one.